caddyhttp/httpserver/middleware.go

```diff
@@ -21,7 +22,7 @@ type (
 
 	// ListenerMiddleware is similar to the Middleware type, except it
 	// chains one net.Listener to the next.
-	ListenerMiddleware func(net.Listener) net.Listener
+	ListenerMiddleware func(caddy.Listener) caddy.Listener
```

Comment:
I can't remember, did we already talk about whether this should be defined and implemented here in the httpserver package or whether it should be in the `caddy` package directly? Would it make sense to register listener wrappers/middlewares for any/all server types or just a specific one (like HTTP) at a time?